packages/node_modules/@ciscospark/spark-core/test/unit/spec/interceptors/rate-limit.js

```diff
@@ -0,0 +1,291 @@
+  Token
+} from '@ciscospark/spark-core';
+import {cloneDeep} from 'lodash';
+describe(`spark-core`, () => {
+  describe(`Interceptors`, () => {
+    describe(`RateLimitInterceptor`, () => {
+
+      function sleep(time) {
+        return new Promise((resolve) => setTimeout(resolve, time));
+      }
+
+      let interceptor, spark;
+      beforeEach(() => {
+        spark = new MockSpark({
+          children: {
+            credentials: Credentials
+          },
+          config: cloneDeep(config)
+        });
+
+        spark.credentials.supertoken = new Token({
+          access_token: `ST1`,
+          token_type: `Bearer`
+        }, {parent: spark});
+
+        interceptor = Reflect.apply(RateLimitInterceptor.create, spark, []);
+      });
+
+      describe(`#isRateLimited`, () => {
+        it(`returns false when URI is not IDbroker URI`, () => {
+          assert.equal(interceptor.isRateLimited(`https://fakeuir.com/testFake`), false);
+        });
+        it(`returns false if the URI is null`, () => {
+          assert.equal(interceptor.isRateLimited(null), false);
+        });
+        it(`returns true when idbroker API is rate limited`, () => {
+          interceptor.setRateLimitExpiry(`https://idbroker.webex.com/horse/v1/myID`, new Date().getTime() * 2);
+          assert.equal(interceptor.isRateLimited(`https://idbroker.webex.com/horse/v1/myID`), true);
+        });
+        it(`returns true when identity API is rate limited`, () => {
+          interceptor.setRateLimitExpiry(`https://identity.webex.com/horse/v1/myID`, new Date().getTime() * 2);
+          assert.equal(interceptor.isRateLimited(`https://identity.webex.com/horse/v1/myID`), true);
+        });
+        it(`returns false when previously rate limited idbroker API expiry time has been met`, () => {
+          interceptor.setRateLimitExpiry(`https://idbroker.webex.com/horse/v1/myID`, 1);
+          return sleep(2000)
+           .then(() => {
+             assert.equal(interceptor.isRateLimited(`https://idbroker.webex.com/horse/v1/myID`), false);
+           });
+        });
+        it(`returns false when previously rate limited identity API expiry time has been met`, () => {
+          interceptor.setRateLimitExpiry(`https://identity.webex.com/horse/v1/myID`, 1);
+          return sleep(2000)
+           .then(() => {
+             assert.equal(interceptor.isRateLimited(`https://identity.webex.com/horse/v1/myID`), false);
+           });
+        });
+      });
+
+      describe(`#getApiName`, () => {
+        it(`returns null when there is no regex match`, () => {
+          assert.equal(interceptor.getApiName(`https://fakeuir.com/testFake`), null);
+        });
+        it(`returns null when the argument is null`, () => {
+          assert.equal(interceptor.getApiName(null), null);
+        });
+        it(`returns idbroker API name if there is a match`, () => {
+          assert.equal(interceptor.getApiName(`https://idbroker.webex.com/horse/v1/myID`), `horse`);
```

Comment:
Is the goal to stop using an entire hostname when we get back a 429 or to stop using the particular endpoint that sent the 429?

Reply:
As discussed in earlier commit messages, the specific API associated with a hostname is limited. The two host names that we've been asked to honor 429s from are the ID broker endpoints.

Reply:
gotcha. sorry, that still seems weird to me, but if that's the plan of record, good enough.